File: Biopython/count.py
# Benchmark the counting of alanine residues in a PDB file

import time
from Bio.PDB import PDBParser

pdb_filepath = "data/1AKE.pdb"
parser = PDBParser()
struc = parser.get_structure("", pdb_filepath)

def count():
    count = 0
    for res in struc.get_residues():
        if res.get_resname() == "ALA":
            count += 1
    return count

start = time.time()
count()
end = time.time()

print(end - start)

File: BioPerl/count.pl
# Benchmark the counting of alanine residues in a PDB file

use Bio::Structure::IO;
use Time::HiRes qw(time);
use strict;

my $pdb_filepath = "data/1AKE.pdb";
my $structio = Bio::Structure::IO->new(-file => $pdb_filepath);
my $struc = $structio->next_structure;

sub count {
    my $c = 0;
    for my $chain ($struc->get_chains) {
        for my $res ($struc->get_residues($chain)) {
            if (substr($res->id, 0, 3) eq "ALA") {
                $c++;
            }
        }
    }
    return $c;
}

my $start = time();
count();
my $end = time();

print $end - $start, "\n";
